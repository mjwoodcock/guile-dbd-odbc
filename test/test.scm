(use-modules (dbi dbi))

(define bad-hnd (dbi-open "odbc" "DSN=completelyinvaliddsn"))
(when (= (car (dbi-get_status bad-hnd)) 0)
  (display "Unexpectedly succeded in opening invalid DSN")
  (newline))

(define hnd (dbi-open "odbc" "DSN=guiledbitest"))
(unless (= (car (dbi-get_status hnd)) 0)
  (display "Unexpected error: ")
  (display (cdr (dbi-get_status hnd)))
  (newline))

(dbi-query hnd "select * from testtable")
(unless (= (car (dbi-get_status hnd)) 0)
  (display "Unexpected error: ")
  (display (cdr (dbi-get_status hnd)))
  (newline))

(define row (dbi-get_row hnd))
(display (dbi-get_status hnd))(newline)
(display row)(newline)

(define row1 (dbi-get_row hnd))
(display (dbi-get_status hnd))(newline)
(display row1)(newline)

