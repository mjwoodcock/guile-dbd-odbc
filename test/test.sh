#!/bin/bash

DBFILE="/tmp/guiledbitest.db"

rm -f $DBFILE
sqlite3 $DBFILE < a.sql
guile test.scm > test.out
if ! diff test.out test.out.expected; then
	echo "Test failed"
	exit 1
fi
