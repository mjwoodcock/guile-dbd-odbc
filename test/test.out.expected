(0 . row fetched)
((a . 1) (b . Hello) (c . 2.3) (d . 2018-02-01) (e . 2018-02-01 09:08:17) (f . 09:08:17) (g . #u8(98 108 111 98 32 100 97 116 97)) (h . #t) (i . 11) (j . 5.6) (k . clob text))
(0 . row fetched)
((a) (b) (c) (d) (e) (f) (g) (h) (i) (j) (k))
